sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/MethodOverloadsShouldBeGrouped.cs

```diff
@@ -274,4 +279,68 @@ protected override void DoSomething(bool b)
 
     }
 
+    public class InterfaceImplementationTogether : IEqualityComparer<int>, IEqualityComparer<string>
+    {
+        public bool Equals(int x, int y) { return true; }
+
+        public int GetHashCode(int obj) { return 0; }
+
+        public bool Equals(string x, string y) { return true; }
+
+        public int GetHashCode(string obj) { return 0; }
+    }
+
+    public interface ITest<TItem>
+    {
+        void Aaa(TItem x);
+        event EventHandler<TItem> Eee;
+        TItem this[TItem index] { get; set; }
+        TItem Ppp { get; set; }
+        void Zzz(TItem x);
+    }
+
+    public class InterfaceImplementationTogetherWithPropertiesAndEvents : ITest<int>, ITest<string>
+    {
+        public event EventHandler<int> Eee;
+        public void Aaa(int x) { }
+        public int this[int index] { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
+        int ITest<int>.Ppp { get; set; }
+        public void Zzz(int x) { }
+
+        event EventHandler<string> ITest<string>.Eee
+        {
+            add { }
+            remove { }
+        }
+        public void Aaa(string x) { }
+        public string this[string index] { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
+        public string Ppp { get; set; }
+        public void Zzz(string x) { }
+    }
+
+    public class ImplicitInterfaceWithEventHandler : ITest<int>
+    {
+        public void Zzz(string x) { } // Noncompliant
```

Comment:
I think we might have a corner case here. If I insert `public void Aaa(string x) { }` right at the beginning of the class (before Zzz), it will be compliant but this is unexpected since it's not grouped with it's overload.

Another note related to this is that if we treat the methods like **Zzz** as noncompliant, the user will have to move them between the interface members in order to get rid of the warning.

Reply:
Both situation are expected and actually documented by this `ImplicitInterfaceWithEventHandler` case. This is FP corner case documented by this comment 
`// C# events are not recognized as parts of interface group, grouping is splitted here`

C# event `Eee` is a corner case, it's not recognized as part of interface group.  And this have some consequences:
- `Zzz(int x)` is not considered as grouped with interface (it's cut off by the `Eee`). So it's standalone and `Zzz(string x)` **is** considered as Noncompliant because of this.
- Inserting `Aaa(string x)` at the beginning is Compliant. Because `Aaa(int x)` is grouped with another interface member `Ppp`. So `Aaa(int x)` **is** grouped by interface and `Aaa(string x)` can be wherever else. This is actually prove that we don't force users to move members between interface members :)

Do you think I should add more comments to document this?

Reply:
Now I understand why it behaves this way. I've missed the group split part. No need for additional comments.